ServerCore/wwwroot/js/pd2022/help.js

```diff
@@ -1,20 +1,98 @@
 window.addEventListener("load", function (event) {
+
+	// Find the author's name so the button can be inserted after it
 	const authordiv = document.getElementsByClassName("byline-div")[0];
 	let childindex = 0;
 	for (let i = 0; i < authordiv.childNodes.length; i++) {
 		if ((authordiv.childNodes[i].nodeType == 3) && (authordiv.childNodes[i].textContent.trim().length > 0)) {
 			childindex = i;
 		}
 	}
-	
+
+	// Create the help button
 	let helpspan = document.createElement("span");
 	helpspan.classList.add("help-link");
-	let helplink = document.createElement("a");
-	helplink.href = "mailto:puzzhunt@microsoft.com?subject=%5B" + 
-		encodeURIComponent(document.getElementsByTagName("h1")[0].innerText.split("\n")[0].trim()) +
-		"%5D&body=The%20more%20details%20you%20add%20here%2C%20the%20more%20helpful%20the%20responses%20you%27ll%20get%20will%20be%21%20%F0%9F%98%87%0D%0A%0D%0A";
+	let helplink = document.createElement("button");
 	helplink.textContent = "click here";
+	helplink.classList.add("help-button");
+	helplink.onclick = clicked;
 	helpspan.append(" (", helplink, " to ask the author for help)");
-	const child = authordiv.childNodes[childindex + 1];
-	authordiv.insertBefore(helpspan, child);
-});
\ No newline at end of file
+
+	// Insert the help button
+	if ((childindex + 1) == authordiv.childNodes.length) {
+		authordiv.insertBefore(helpspan, null);
+	}
+	else {
+		const child = authordiv.childNodes[childindex + 1];
+		authordiv.insertBefore(helpspan, child);
+	}
+});
+
+async function clicked(event) {
+
+	// If the URL has the necessary parameters to call the team name API, do it
+	if (window.location.href.split('?').length > 1) {
+		let params = window.location.href.split('?')[1];
+		if ((!params.includes("eventId")) || (!params.includes("puzzleId")) || (!params.includes("teamPassword"))) {
+			backup();
+		} else {
+			if ((params.split('&')[0].split('=')[1].length == 0) || (params.split('&')[1].split('=')[1].length == 0) || (params.split('&')[2].split('=')[1].length == 0)) {
```

Comment:
Consider using UrlSearchParams here. See usage at https://www.w3docs.com/snippets/javascript/how-to-get-url-parameters.html, and note the use of window.location.search to jump right to the queryString.

Reply:
Adding in next commit